src/dscanner/analysis/autofix.d

```diff
@@ -0,0 +1,196 @@
+module dscanner.analysis.autofix;
+
+import std.algorithm : filter, findSplit;
+import std.conv : to;
+import std.functional : toDelegate;
+import std.stdio;
+
+import dparse.lexer;
+import dparse.rollback_allocator;
+import dparse.ast : Module;
+
+import dsymbol.modulecache : ModuleCache;
+
+import dscanner.analysis.base : AutoFix, AutoFixFormatting, BaseAnalyzer, Message;
+import dscanner.analysis.config : StaticAnalysisConfig;
+import dscanner.analysis.run : analyze, doNothing;
+import dscanner.utils : readFile, readStdin;
+
+private void resolveAutoFixes(
+	ref Message message,
+	string fileName,
+	ref ModuleCache moduleCache,
+	scope const(Token)[] tokens,
+	const Module m,
+	const StaticAnalysisConfig analysisConfig,
+	const AutoFixFormatting overrideFormattingConfig = AutoFixFormatting.invalid
+)
+{
+	resolveAutoFixes(message.checkName, message.autofixes, fileName, moduleCache,
+		tokens, m, analysisConfig, overrideFormattingConfig);
+}
+
+private void resolveAutoFixes(string messageCheckName, AutoFix[] autofixes, string fileName,
+	ref ModuleCache moduleCache,
+	scope const(Token)[] tokens, const Module m,
+	const StaticAnalysisConfig analysisConfig,
+	const AutoFixFormatting overrideFormattingConfig = AutoFixFormatting.invalid)
+{
+	import core.memory : GC;
+	import dsymbol.conversion.first : FirstPass;
+	import dsymbol.conversion.second : secondPass;
+	import dsymbol.scope_ : Scope;
+	import dsymbol.semantic : SemanticSymbol;
+	import dsymbol.string_interning : internString;
+	import dsymbol.symbol : DSymbol;
+	import dscanner.analysis.run : getAnalyzersForModuleAndConfig;
+
+	const(AutoFixFormatting) formattingConfig =
+	overrideFormattingConfig is AutoFixFormatting.invalid
+	? analysisConfig.getAutoFixFormattingConfig()
+	: overrideFormattingConfig;
+
+	scope first = new FirstPass(m, internString(fileName), &moduleCache, null);
+	first.run();
+
+	secondPass(first.rootSymbol, first.moduleScope, moduleCache);
+	auto moduleScope = first.moduleScope;
+	scope(exit) typeid(DSymbol).destroy(first.rootSymbol.acSymbol);
+	scope(exit) typeid(SemanticSymbol).destroy(first.rootSymbol);
+	scope(exit) typeid(Scope).destroy(first.moduleScope);
+
+	GC.disable;
+	scope (exit)
+	GC.enable;
+
+	foreach (BaseAnalyzer check; getAnalyzersForModuleAndConfig(fileName, tokens, m, analysisConfig, moduleScope))
+	{
+		if (check.getName() == messageCheckName)
+		{
+			foreach (ref autofix; autofixes)
+				autofix.resolveAutoFixFromCheck(check, m, tokens, formattingConfig);
+			return;
+		}
+	}
+
+	throw new Exception("Cannot find analyzer " ~ messageCheckName
+	~ " to resolve autofix with.");
+}
+
+///
```

Comment:
What's the point of these empty ddoc comments?

Reply:
I used them to mark which functions are used outside the module, so that it's easier to refactor in the future

Reply:
Use a normal comment then, not something that the compiler actually recognizes and interprets. This is confusing for anyone looking at the code.

Reply:
Will fix